src/PrebidMobile/PrebidMobile/AdUnits/AdUnit.swift

```diff
@@ -16,184 +16,233 @@
+
+        if (prebidConfigId.isEmpty || (prebidConfigId.trimmingCharacters(in: CharacterSet.whitespaces)).count == 0) {
             completion(ResultCode.prebidInvalidConfigId)
             return
         }
-        if(Prebid.shared.prebidServerAccountId.isEmpty || (Prebid.shared.prebidServerAccountId.trimmingCharacters(in: CharacterSet.whitespaces)).count == 0){
+        if (Prebid.shared.prebidServerAccountId.isEmpty || (Prebid.shared.prebidServerAccountId.trimmingCharacters(in: CharacterSet.whitespaces)).count == 0) {
             completion(ResultCode.prebidInvalidAccountId)
             return
         }
-        if(isInitialCallMade == false){
+        if (isInitialCallMade == false) {
             //the publisher called the fetch demand 1st fire the timer
              isInitialCallMade = true
             //start the timer only if the refresh timer is valided & set
-            if(refreshTime! > 0.0){
+            if (refreshTime! > 0.0) {
                 self.timerClass?.start(autoRefreshMillies: refreshTime!)
             }
         }
-       
+
         didReceiveResponse = false
         timeOutSignalSent = false
         self.closure = completion
         adServerObject = adObject
-        let manager:BidManager = BidManager(adUnit: self)
-        
-        manager.requestBidsForAdUnit() { (bidResponse, resultCode) in
+        let manager: BidManager = BidManager(adUnit: self)
+
+        manager.requestBidsForAdUnit { (bidResponse, resultCode) in
             self.didReceiveResponse = true
-            if(bidResponse != nil){
-                if(!self.timeOutSignalSent){
+            if (bidResponse != nil) {
+                if (!self.timeOutSignalSent) {
                         Utils.shared.validateAndAttachKeywords (adObject: adObject, bidResponse: bidResponse!)
                         completion(resultCode)
                 }
-                
+
             } else {
-                if(!self.timeOutSignalSent){
+                if (!self.timeOutSignalSent) {
                     completion(resultCode)
                 }
             }
         }
-        
-        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(.PB_Request_Timeout) , execute: {
-            if(!self.didReceiveResponse){
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(.PB_Request_Timeout), execute: {
+            if (!self.didReceiveResponse) {
                 self.timeOutSignalSent = true
                 completion(ResultCode.prebidDemandTimedOut)
-                
+
             }
         })
     }
-    
-    
-    
-    var userKeywords:[String: Array<String>] {
-        Log.info("user keywords are \(customKeywords)")
-        return customKeywords
+
+    var userKeywords: [String: [String]] {
+        Log.info("user keywords are \(customUserKeywords)")
+        return customUserKeywords
     }
-    
+
+    var invKeywords: [String: [String]] {
```

Comment:
This is a new API addition... Were is the requirements for this? We need to have the feature created before coding in. Please create a feature request first and have the products approval before adding in the code for the same

Reply:
@ppuviarasu , here is the feature request: #165

Reply:
As @andrew suggested, please split the PR. Keep the feature PR separate

Reply:
Hi @ppuviarasu , sure. I have removed it and add it in a new PR:
https://github.com/prebid/prebid-mobile-ios/pull/166

Thanks